astropy/time/core.py

```diff
@@ -282,6 +283,8 @@ def _get_time_fmt(self, val, val2, format, scale):
             try:
                 return FormatClass(val, val2, scale, self.precision,
                                    self.in_subfmt, self.out_subfmt)
+            except UnitConversionError:
```

Comment:
I double checked, and there are a few tests that fail without this line.  For example `TestTimeQuantity.test_invalid_quantity_input` fails, because it tests instantiating a `Time` object from a quantity in units of meters.  This should definitely result in a unit conversion error, instead of just being ignored (instead you end up getting an error "Input values did not match the format class jd", which is true in a sense, but it's not as useful as a unit error would be here.